staging/src/k8s.io/client-go/scale/client.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scale
+
+import (
+	"fmt"
+
+	autoscaling "k8s.io/api/autoscaling/v1"
```

Comment:
this design puts us into a strange situation when we want to add /scale support to CRDs:

1. we don't have Scale in meta/v1 because it is not general enough. Instead we put it into k8s.io/api.
2. apiextensions-apiserver is not kube-specific enough to move its types into k8s.io/api, instead we keep it logically separate.
3. client-go cannot import k8s.io/apiextensions-apiserver due to an import cycle.

Following 2, we would like to have meta/v1.Scale or apiextensions/v1beta1.Scale. But 1 and 3 stops us from that. Technically, we can import autoscaling/v1 in k8s.io/apiextensions-apiserver and use that Scale type for CRDs. But this conflicts with 2.